distributed/spill.py

```diff
@@ -1,40 +1,50 @@
 from __future__ import annotations
 
+import logging
 from collections.abc import Hashable, Mapping
+from distutils.version import LooseVersion
 from functools import partial
 from typing import Any
 
-from zict import Buffer, File, Func
+import zict
+from typing_extensions import Literal
 
 from .protocol import deserialize_bytes, serialize_bytelist
 from .sizeof import safe_sizeof
 
+logger = logging.getLogger(__name__)
 
-class SpillBuffer(Buffer):
+
+class SpillBuffer(zict.Buffer):
     """MutableMapping that automatically spills out dask key/value pairs to disk when
-    the total size of the stored data exceeds the target
+    the total size of the stored data exceeds the target. If max_spill is provided the
+    key/value pairs won't be spilled once this threshold has been reached.
     """
 
-    spilled_by_key: dict[Hashable, int]
-    spilled_total: int
+    def __init__(
+        self, spill_directory: str, target: int, max_spill: int | Literal[False] = False
+    ):
+        if (
+            max_spill and LooseVersion(zict.__version__) <= "2.0"
+        ):  # this is not giving warning now because 2.0.0 > 2.0
+            raise ValueError("zict > 2.0 required to set max_weight")
```

Comment:
Once we change to `LooseVersion(zict.__version__) <= "2.0.0"` things work, the problem was that if zict version is 2.0.0,  `LooseVersion(zict.__version__) <= "2.0"` returns False. For now, until we get zict merged I'll leave `2.0` otherwise I can't test the code. Will fix this later by using `LooseVersion(zict.__version__) <= "2.0.0"`